pkgs/servers/mastodon/default.nix

```diff
@@ -28,12 +27,12 @@ stdenv.mkDerivation rec {
     pname = "${pname}-modules";
     inherit src version;
 
-    yarnOfflineCache = fetchYarnDeps {
-      yarnLock = "${src}/yarn.lock";
+    yarnOfflineCache = callPackage ./yarn.nix {
+      inherit src;
       hash = yarnHash;
     };
 
-    nativeBuildInputs = [ fixup-yarn-lock nodejs-slim yarn mastodonGems mastodonGems.wrappedRuby brotli ];
+    nativeBuildInputs = [ nodejs-slim yarn-berry mastodonGems mastodonGems.wrappedRuby brotli python3 ];
```

Comment:
Why do we need such an old python version? Maybe distutils? Can we add it in an env instead?

Reply:
If use the latest versions of Pythons a build error occurs:
```
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info it worked if it ends with ok
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info using node-gyp@10.0.1
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info using node@20.17.0 | linux | x64
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info find Python using Python version 3.12.5 found at "/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3"
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn /nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args [
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '/build/source/node_modules/node-gyp/gyp/gyp_main.py',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args 'binding.gyp',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-f',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args 'make',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-I',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '/build/source/node_modules/utf-8-validate/build/config.gypi',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-I',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '/build/source/node_modules/node-gyp/addon.gypi',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-I',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '/nix/store/a3rj28n8mblqs26nrn7wzl4m8hda8rkk-nodejs-slim-22.8.0/include/node/common.gypi',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dlibrary=shared_library',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dvisibility=default',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dnode_root_dir=/nix/store/a3rj28n8mblqs26nrn7wzl4m8hda8rkk-nodejs-slim-22.8.0',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dnode_gyp_dir=/build/source/node_modules/node-gyp',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dnode_lib_file=/nix/store/a3rj28n8mblqs26nrn7wzl4m8hda8rkk-nodejs-slim-22.8.0/$(Configuration)/node.lib',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dmodule_root_dir=/build/source/node_modules/utf-8-validate',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Dnode_engine=v8',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '--depth=.',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '--no-parallel',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '--generator-output',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args 'build',
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args '-Goutput_dir=.'
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args ]
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR <string>:17: SyntaxWarning: invalid escape sequence '\.'
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn make
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT make: Entering directory '/build/source/node_modules/utf-8-validate/build'
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT   CXX(target) Release/obj.target/validation/src/validation.o
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT   CXX(target) Release/obj.target/validation/deps/is_utf8/src/is_utf8.o
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT   SOLINK_MODULE(target) Release/obj.target/validation.node
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT   COPY Release/validation.node
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDOUT make: Leaving directory '/build/source/node_modules/utf-8-validate/build'
➤ YN0000: │ utf-8-validate@npm:6.0.4 STDERR gyp info ok
```

Reply:
I don't see an error in the logs and locally it build also fine. Am I missing something?

Reply:
Can be seen with the additional parameter `--inline-builds`.

Reply:
Try adding setuptools. That often fixed issues with pre-gyp and the like.

Reply:
The `setuptools` package does not resolve the error.

Reply:
The error will be displayed if you build with the following command:
```
  yarn install --immutable --immutable-cache --inline-builds
```